shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -334,51 +309,60 @@ v8::MaybeLocal<v8::Value> PassValueToOtherContext(
   }
 }
 
-v8::Local<v8::Value> ProxyFunctionWrapper(
-    context_bridge::RenderFrameFunctionStore* store,
-    size_t func_id,
-    bool support_dynamic_properties,
-    gin_helper::Arguments* args) {
+void ProxyFunctionWrapper(const v8::FunctionCallbackInfo<v8::Value>& info) {
   TRACE_EVENT0("electron", "ContextBridge::ProxyFunctionWrapper");
+  CHECK(info.Data()->IsObject());
+  v8::Local<v8::Object> data = info.Data().As<v8::Object>();
+  bool support_dynamic_properties = false;
+  gin::Arguments args(info);
   // Context the proxy function was called from
-  v8::Local<v8::Context> calling_context = args->isolate()->GetCurrentContext();
-  // Context the function was created in
-  v8::Local<v8::Context> func_owning_context =
-      std::get<1>(store->functions()[func_id]).Get(args->isolate());
+  v8::Local<v8::Context> calling_context = args.isolate()->GetCurrentContext();
+
+  // Pull the original function and its context off of the data private key
+  v8::MaybeLocal<v8::Value> sdp_value =
+      GetPrivate(calling_context, data, "support_dynamic_properties");
+  v8::MaybeLocal<v8::Value> maybe_func =
+      GetPrivate(calling_context, data, "bridged_fn");
+  v8::Local<v8::Value> func_value;
+  if (sdp_value.IsEmpty() || maybe_func.IsEmpty() ||
+      !gin::ConvertFromV8(args.isolate(), maybe_func.ToLocalChecked(),
+                          &support_dynamic_properties) ||
```

Comment:
am i blind or is this trying to convert `maybe_func` to a bool?

Reply:
Wat, how did this pass tests lol